### PR TITLE
Register miopen_batch_norm for lowering

### PR DESCRIPTION
**Pull request by jataylo**
Registers aten::miopen_batch_norm to enable lowering into inductor.

```
python test_torchinductor_opinfo.py -v TestInductorOpInfoCUDA.test_comprehensive_nn_functional_batch_norm_cuda_float32 
test_comprehensive_nn_functional_batch_norm_cuda_float32 (__main__.TestInductorOpInfoCUDA) ... ok

python test_torchinductor_opinfo.py -v TestInductorOpInfoCUDA.test_comprehensive_nn_functional_instance_norm_cuda_float32 
test_comprehensive_nn_functional_instance_norm_cuda_float32 (__main__.TestInductorOpInfoCUDA) ... ok

python test_torchinductor.py -v CudaTests.test_batch_norm_2d_dynamic_shapes_cuda
ok
```

Originally I faced issues trying to implement an equivalent method in `decompositions.py` to `cudnn_batch_norm` but after tracing the issues that arose I found that the schema for the miopen/cudnn method differ.
https://github.com/ROCmSoftwarePlatform/pytorch/blob/65c0bb4ce924ab91f033e97b3a56c1f092cf5a57/torch/csrc/utils/schema_info.cpp#L284

This solution proposes simply calling into native_batch_norm as we share the same signature

### CODE REVIEW

**Comment by jataylo** [comment]
Looking at https://github.com/ROCmSoftwarePlatform/pytorch/blob/master/aten/src/ATen/native/miopen/BatchNorm_miopen.cpp we may want to add similar conditional logic to miopen_batch_norm in decompositions.py 

**cudnn_batch_norm:**
```
# Cudnn return running mean and variance when training is True
if training:
    return (a, b, c, input.new_zeros((0,), dtype=torch.uint8))
return (
    a,
    weight.new_zeros((0,)),
    weight.new_zeros((0,)),
    input.new_zeros((0,), dtype=torch.uint8),
)
```

**Aten/native/miopen/BatchNorm_miopen.cpp**
```
  if (training) {
    int64_t num_features = input_t.size(1);
    save_mean = at::empty({ num_features }, weight_t.options());
    save_var = at::empty({ num_features }, weight_t.options());
    MIOPEN_CHECK(miopenBatchNormalizationForwardTraining(
      handle, mode, &one, &zero,
      idesc.desc(), input->data_ptr(),
      idesc.desc(), output->data_ptr(),
      wdesc.desc(),
      weight->data_ptr(),
      bias->data_ptr(),
      exponential_average_factor,
      at::maybe_data_ptr(running_mean),
      at::maybe_data_ptr(running_var),
      epsilon,
      save_mean.data_ptr(),
      save_var.data_ptr()));
  } else {
    save_mean = at::empty({0}, weight_t.options());
    save_var = at::empty({0}, weight_t.options());
    MIOPEN_CHECK(miopenBatchNormalizationForwardInference(
      handle, mode, &one, &zero,
      idesc.desc(), input->data_ptr(),
      idesc.desc(), output->data_ptr(),
      wdesc.desc(),
      weight->data_ptr(),
      bias->data_ptr(),
      running_mean->data_ptr(),
      running_var->data_ptr(),
      epsilon));
  }
```

**Proposed change:**
```
@aten.miopen_batch_norm.default.py_impl(DispatchKey.Autograd)
@register_decomposition(aten.miopen_batch_norm)
def miopen_batch_norm(
    input: Tensor,
    weight: Tensor,
    bias: Optional[Tensor],
    running_mean: Optional[Tensor],
    running_var: Optional[Tensor],
    training: bool,
    exponential_average_factor: float,
    epsilon: float,
):
    a,b,c = aten.native_batch_norm(
        input,
        weight,
        bias,
        running_mean,
        running_var,
        training,
        exponential_average_factor,
        epsilon,
    )
    
    if training:
        return (a, b, c)
    return (
        a,
        weight.new_zeros((0,)),
        weight.new_zeros((0,)),
    )
```

**Comment by jataylo** [comment]
Closing, we likely won't take this for now due to perf.
